### DISCUSSION TITLE
How to get Roles in channel and get Users ID in roles

### DISCUSSION THREAD

**Discussion by GalsCressendo** [bwmarrin/discordgo#1239] [Q&A]
I want to create a bot to dm user, but only user with certain roles. I have read https://pkg.go.dev/github.com/bwmarrin/discordgo#section-readme , but haven't how to get the roles from the server and then get all the users ID from that role.
Thankyou in advance

**Comment by Nv7-GitHub** [comment]
You could get all users and check if they have that role (use state for this, make sure to have guild intents)
